### PR TITLE
Add basic user handling

### PR DESCRIPTION
**Pull request by sjpb**
Provides a basic IAM capability via ansible. See [ansible/roles/basic_users/README.md](https://github.com/stackhpc/ansible-slurm-appliance/blob/feature/basic_users/ansible/roles/basic_users/README.md) for features/why this isn't totally straightforward.

Unlike the p3-appliances implementation, the datastructure used to define users etc is passed through to ansible's user module, so supports anything that module does.

Name chosen as `local_users` is already used for `podman` etc system users.

### CODE REVIEW

File: ansible/roles/basic_users/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    label: "{{ item.name }}"
+  when:
+    - "item.state | default('present') == 'absent'"
+  
+- name: Create users and generate public keys
+  user: "{{ basic_users_userdefaults | combine(item) | remove_keys(['public_key']) }}"
+  loop: "{{ basic_users_users }}"
+  loop_control:
+    label: "{{ item.name }} [{{ item.state | default('present') }}]"
+  register: basic_users_info
```
Comment (jovial) [inline ansible/roles/basic_users/tasks/main.yml:16]: Could you compute this once on localhost?

File: ansible/roles/basic_users/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    - basic_users_manage_homedir
+  run_once: true
+
+- name: Write generated public key as authorized for SSH access
+  authorized_key:
+    user: "{{ item.name }}"
+    state: present
+    manage_dir: no
+    key: "{{ item.ssh_public_key }}"
+  loop: "{{ hostvars[ansible_play_hosts | first].basic_users_info.results }}"
```
Comment (jovial) [inline ansible/roles/basic_users/tasks/main.yml:38]: I think `basic_users_info` is set as a fact on all hosts, right?

File: ansible/roles/basic_users/library/terminate_user_sessions.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2021, Steve Brasier <steveb@stackhpc.com>
+# Apache V2 licence
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
```
Comment (jovial) [inline ansible/roles/basic_users/library/terminate_user_sessions.py:8]: Update or remove the docs.

File: ansible/roles/basic_users/filter_plugins/remove.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class FilterModule(object):
+
+    def filters(self):
+        return {
+            'remove_keys': self.remove_keys
+        }
+
+    def remove_keys(self, orig_dict, keys_to_remove):
```
Comment (jovial) [inline ansible/roles/basic_users/filter_plugins/remove.py:13]: remove suggests that the original dict is modified. Would filter_keys be a better name?

File: ansible/roles/basic_users/tasks/main.yml (hunk 2 above)
Comment (sjpb) [reply ansible/roles/basic_users/tasks/main.yml:38]: Yes but the key is only generated on the first one (by design), so it is only present in the facts for that host.

**Review by jovial — APPROVED** [review]
All comments have now been addressed, looks good :)

**Pull request merged**